packages/evm/contracts/incentives/AERC20PeggedIncentive.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.24;
+
+import {LibPRNG} from "@solady/utils/LibPRNG.sol";
+import {SafeTransferLib} from "@solady/utils/SafeTransferLib.sol";
+
+import {ACloneable} from "contracts/shared/ACloneable.sol";
+import {BoostError} from "contracts/shared/BoostError.sol";
+
+import {ABudget} from "contracts/budgets/ABudget.sol";
+import {AIncentive} from "./AIncentive.sol";
+
+/// @title ERC20 AIncentive
+/// @notice A simple ERC20 incentive implementation that allows claiming of tokens
+abstract contract AERC20PeggedIncentive is AIncentive {
+    using LibPRNG for LibPRNG.PRNG;
+    using SafeTransferLib for address;
+
+    /// @notice A mapping of address to claim status
+    mapping(address => bool) public claimed;
+
+    uint256 public totalClaim;
+
+    /// @notice The limit (max claims, or max entries for raffles)
+    function limit() external virtual returns (uint256);
```

Comment:
It's the max spend for the incentive, since we don't know what the reward amount will be in reward token we need to use the limit similarly to a variable incentive. There's some info in the PR description too but let me update this comment. LMK if you see any other comments that are off